skimage/feature/_canny.py

```diff
@@ -85,7 +83,7 @@ def _preprocess(image, mask, sigma, mode, cval):
         # the function to the mask (which gets you the fraction of the
         # pixel data that's due to significant points)
         bleed_over = gaussian(mask.astype(float_type, copy=False),
-                              **gaussian_kwargs) + np.finfo(float_type).eps
+                              sigma=1, mode='nearest') + np.finfo(float_type).eps
```

Comment:
Changing mode from `constant` to `nearest` is not necessary to fix this issue, and is better suited in `canny` function signature.

Reply:
Without it, the results with a full mask do not match the unmasked version.